### ISSUE TITLE
Please add a cmd option (/ini)

### ISSUE THREAD

**Issue by Martello124c41** [WombatFromHell/OriginSteamOverlayLauncher#6]
This is more of an enhancement request.

Could you add a cmd option where OSOL can find its ini file (or create it)?

I tried OSOL together with GOG galaxy and placing OSOL in the folder of the game didn't work.
Well the game started but the overlay of GOGgalaxy didn't work. Moving OSOL into the folder of the launcher changed that.
I used LauncherOnly and the cmd-line in the shortcut that gog created to run the game with gog.
Thanks for your efforts.

**Comment by WombatFromHell** [comment]
Hmm, that sounds like a bug. I'll investigate and get back to you.

If it turns out I need to do some special things for GOG Galaxy I'll relabel as Enhancement.

**Comment by WombatFromHell** [comment]
I'm still investigating, but I will say a few things about GOG Galaxy given that its overlay support is similar to Battle.net in several ways:

- Make sure you're using the right launcher executable target (GalaxyClient.exe in this case)
- If a particular title has a launcher of its own make sure to target the right game executable (the one the game launcher actually loads)
- Make certain that Galaxy isn't running before OSOL attempts to launch it (OSOL tries to do this for you but it can be spotty on rare occasions)

From my initial testing so far it seems that OSOL isn't causing the issue itself. It could be some conflict between the game in question, GOG Galaxy's overlay injection method, and Steam's injection method. I have no way to diagnose injection methods with OSOL because I don't memory hook or inject anything - OSOL is just a simple launcher with some quality of life enhancements for launching launchers through Steam.

Could you give me a clue as to which game you're having these issues with?

**Comment by Martello124c41** [comment]
That game would be Battle Chaser: Nightwar
This is the ini i use 
[OriginSteamOverlayLauncher.txt](https://github.com/WombatFromHell/OriginSteamOverlayLauncher/files/1444256/OriginSteamOverlayLauncher.txt)

I also think it is the game that causes the problem.
Witcher 3 for example runs just fine.

**Comment by WombatFromHell** [comment]
I'm testing against Witcher 3: GOTY Edition using your launcher args with the Witcher 3 game id and path [like so](https://gist.github.com/WombatFromHell/bb0005eb7c8cbca0366575d1c5a4fbce#file-originsteamoverlaylauncher-ini).

This works perfectly no matter where I place the OSOL exe and ini file. The GOG Galaxy Overlay, RTSS overlay, and Steam overlay all come up properly with their respective shortcuts and function normally inside and outside the game. Galaxy reflects the time played and cloud syncs before being closed by OSOL after the configured timeout, as normal.

If you have more specific steps on how to reproduce what you're seeing please do.

**Comment by Martello124c41** [comment]
You are right, as long as i don't run OSOL in the game folder, everything is working.
I assume, because i put some "" on the /path-parameter the Overlay in Battle Chasers is working as well with it started in the game folder.
Otherwise, i don't know why it's now working and not before.

Anyway with OSOL in the games folder cloud saves are still not working but thats a feature of GOG Galaxy. In the log at C:\ProgramData\GOG.com\Galaxy\logs\GalaxyClient.log it details why:

2017-11-05 22:51:24.141 [Warning][ (0)]: Aborting cloud storage sync as at least one process is running inside game's folder (D:\Games\GOG Galaxy\Games\The Witcher 3 Wild Hunt GOTY). Detected executing processes:
2017-11-05 22:51:24.141 [Information][ (0)]: PID: 2420, path: D:\Games\GOG Galaxy\Games\The Witcher 3 Wild Hunt GOTY\bin\x64\OriginSteamOverlayLauncher.exe

So in summary i have a solution and don't need the cmd-option.
Thanks for your help. :)

**Comment by WombatFromHell** [comment]
Good catch on that Galaxy related syncing issue. I'll add a section to the project wiki about it in case anyone else has problems.

If you have any other problems I can help with don't hesitate to open another ticket.
